app/src/main/java/com/github/HumanLearning2021/HumanLearningApp/presenter/LearningPresenter.kt

```diff
@@ -19,8 +37,17 @@ class LearningPresenter(
         Log.d("displayNextPicture", "previous : $previousCategory, curr : $rndCat")
         previousCategory = rndCat
 
-        databaseService.getPicture(rndCat!!)?.displayOn(activity, view)
-        view.contentDescription = rndCat.name
+        val nextPicture = databaseService.getPicture(rndCat!!)
+        nextPicture!!.displayOn(activity, view)
+        view.contentDescription = rndCat!!.name
         view.invalidate()
     }
+
+
+    suspend fun displayTargetPicture(activity: Activity, view: ImageView, categoryName: String){
+        if(learningMode == LearningMode.PRESENTATION)
+            databaseService.getPicture(databaseService.getCategory(categoryName)!!)?.displayOn(activity, view)
+        else
+           databaseService.getRepresentativePicture(categoryName)?.displayOn(activity, view)
```

Comment:
This should be flipped no ? (presentation : match identical images, representation: match images that have the same category). (also maybe use a when statement, it's more adapted to an enum)

Reply:
Yes for the when. Presentation is identical images indeed, so representation is against representatives of the category. Or am I confused?

Reply:
I'm not sure, but I think you're overcomplicating things here. The  target image views (on which you drag the image to sort) will always display the category representatives. What changes between presentation and representation is from which images you sort. In presentation you only sort the representatives, and in representation, you use all the images in the category. I should have explained that more clearly, my bad. Is it clear now?

Reply:
Oh okay, I did it the other way around: in my code what changes between presentation and representation is the target images, whereas you're talking about the image to sort.

Reply:
I can do it your way though, no issue with that.

Reply:
No problem. It's just that it was done this way in the physical version, so I'd like to keep it as close as possible to that.

Reply:
Yes definitely. I'm not sure I can finish this tonight though, I'll DM you when it's done.